p4rt_client/p4rt_client.go

```diff
@@ -346,6 +348,51 @@ func (p *P4RTClientStream) GetPacket(minSeqNum uint64) (uint64, *P4RTPacketInfo,
 	return p.pktCounters.RxPktCntr, pktInfo, nil
 }
 
+func (p *P4RTClientStream) GetPacketsWithTimeout(minSeqNum uint64, timeout time.Duration) (uint64, []*P4RTPacketInfo, error) {
+
+	done := make(chan struct{})
+	defer close(done)
+	var failed atomic.Bool
+
+	// routine returns directly if 'done' else updates 'failed'
+	// and signals pktCond.
+	go func() {
+		select {
+		case <-done:
+			return
+		case <-time.After(timeout):
+			failed.Store(true)
+			p.pktCond.Signal()
```

Comment:
Signal() wakes "one" Go routine (as per documentation). In case both GetPacket() and GetPacketsWithTimeout() are called (they should not be, and thus a comment on top of both functions is necessary i.e. the user MUST use one or the other - please add this comment on top of both functions), should we use Broadcast() instead? Broadcast() should replace both .Signal() calls in GetPacket() and GetPacketsWithTimeout()

Reply:
I think we should call Broadcast().

Updated to Broadcast() and also added an additional comment.